### PR TITLE
Add pure python date annotator

### PR DESCRIPTION
**Pull request by nathanathan**
The new annotator is designed to replace the jvm-nlp annotator for the reasons discussed here: https://www.pivotaltracker.com/story/show/146162419. It uses the SpaCy NER and regular expressions to detect dates and it uses the dateparse library, along with some custom code, to parse them into datetime ranges.

This also adds some usage instructions to the documentation and adds dictionary return values to the Annotator.annotate method.

### CODE REVIEW

File: tests/annotator/test_date_annotator.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+from __future__ import absolute_import
+import unittest
+import datetime
+from epitator.annotator import AnnoDoc
+from epitator.date_annotator import DateAnnotator
+
+
+class DateAnnotatorTest(unittest.TestCase):
```
Comment (Broham) [inline tests/annotator/test_date_annotator.py]: Should we add a test for dates with dashes (or slashes etc.)?  I'm seeing some weird behavior - I added the following test:

```    
def test_dashes(self):
        text = 'Adenoviruses, first seen between 2010-1-1 and 2010-1-2'
        doc = AnnoDoc(text)
        doc.add_tier(self.annotator)
        self.assertEqual(
            doc.tiers['dates'].spans[0].datetime_range,
            [datetime.datetime(2010, 1, 1),
            datetime.datetime(2010, 1, 2)])
```

and i'm getting the results:

```
AssertionError: Lists differ: [datetime.datetime(2010, 1, 2,... != [datetime.datetime(2010, 1, 1,...

First differing element 0:
2010-01-02 00:00:00
2010-01-01 00:00:00

- [datetime.datetime(2010, 1, 2, 0, 0), datetime.datetime(2010, 1, 3, 0, 0)]
?                             ^                                    ^

+ [datetime.datetime(2010, 1, 1, 0, 0), datetime.datetime(2010, 1, 2, 0, 0)]
?                             ^                                    ^
```

Comment (nathanathan) [inline tests/annotator/test_date_annotator.py]: Yes, thanks, I'm adding your test and fixing the bugs it uncovered.

File: epitator/date_annotator.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                                     date_spans], prefer='text_length'):
+            # Parse the span text into one or two components depending on
+            # whether it contains multiple dates for specifying a range.
+            if isinstance(date_span, ra.MatchSpan) and\
+               date_span.match_name == 'date_range':
+                range_components = [span.text
+                                    for span in date_span.base_spans[0::2]]
+            else:
+                range_components = re.split(
+                    r"\b(?:and|to|through|until)\b", date_span.text)
```
Comment (nathanathan) [inline epitator/date_annotator.py]: I'm refactoring to use a shared regex for both matches.

File: epitator/date_annotator.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+of December 1st to the end of December.
+"""
+from __future__ import absolute_import
+from dateutil.parser import parser
+from .annotator import AnnoDoc, Annotator, AnnoTier, AnnoSpan
+from .spacy_annotator import SpacyAnnotator
+from . import result_aggregators as ra
+from dateparser.date import DateDataParser
+from dateutil.relativedelta import relativedelta
+import requests
```
Comment (nathanathan) [inline epitator/date_annotator.py]: Good catch

File: epitator/date_annotator.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+                   for span in date_group_spans):
+                extended_span = date_group_spans[0].extended_through(
+                    date_group_spans[-1])
+                if date_to_datetime_range(extended_span.text) != None:
+                    grouped_date_spans.append(extended_span)
+        # Find date ranges by looking for joiner words between dates.
+        date_range_spans = ra.follows([
+            date_spans,
+            [t_span for t_span in doc.tiers['spacy.tokens'].spans
+             if re.match(r"(to|through|\-|between|until|and)$", t_span.text)],
```
Comment (nathanathan) [inline epitator/date_annotator.py]: It could help if the sentence is in all caps.

**Pull request merged**